### PR TITLE
Add GitLab on DC/OS blog post and tutorial.

### PR DESCRIPTION
**Pull request by ssk2**

### CODE REVIEW

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+date: 2016-09-16
+author: Sunil Shah, Mesosphere
+category: services
+description: Create a continuous delivery pipeline using the new GitLab package for DC/OS.
+layout: article.jade
+collection: posts
+lunr: true
+---
+
+We've been working with the team at GitLab for a few months now to bring GitLab as an easily installable package to DC/OS. This is a feature that many in the community have requested and we're happy to finally merge this package to the Universe and make it available for all!
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: This isn't part of DC/OS community, let's not use words like `we`. Maybe `you can now use GitLab easily with DC/OS` or something like that. Need to make sure the community is aware that this isn't a part of DC/OS but an application you run on top of it.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+If you're unfamiliar with GitLab, it's an open source developer tool that allows you to host git repositories, review code, track issues, host Docker images and perform continuous integration. (In this very first version of the integration, GitLab uses Jenkins for continuous integration, but we are working to bring GitLab CI to DC/OS too!)
+
+Using GitLab on DC/OS now allows you to co-locate all of the tools you need for developers on one easy to manage cluster. Just as with any Universe package, you can robustly install several side by side instances of GitLab to provide segregated instances for each of your development teams. Alternatively, you can just as easily install GitLab in a highly available configuration that many teams use concurrently.
+
+In this post, we'll run through a quickstart installation of GitLab to show how straightforward it is to set up an end to end continuous delivery pipeline using GitLab to host both your code and Docker images, and Jenkins to run a job that builds a Docker image on DC/OS and deploys it to Marathon.
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Please move all this into an actual tutorial. Discovery on blog posts and maintenance is a nightmare. Then you can link to the tutorial in docs and we'll bring it along and update every version.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+<img src="/assets/images/blog/2016-09-16-gitlab/gitlab-cd.png" alt="Continuous Delivery with GitLab" /> *Continuous Delivery with GitLab*
+
+This quickstart installation uses the single node version of GitLab that includes an installation of Postgres and Redis in the same container.
+
+The instructions below use a pinned hostname constraint to success the application is always restarted on the same host by Marathon. This allows it to get back to its data but means that you could lose data if that agent goes down. We recommend checking out the production installation instructions on alternative options and considerations.
+
+# Prerequisites
+
++ DC/OS 1.8 or later
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Please link to the installation docs here.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
++ [Marathon-lb](https://dcos.io/docs/1.8/usage/service-discovery/marathon-lb/usage/) must be installed and running
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Drop the domain.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
++ [Jenkins](https://docs.mesosphere.com/1.8/usage/service-guides/jenkins/) must be installed and running
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Drop the domain.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
++ An available hostname configured to point to the public agent(s) where Marathon-lb is running (e.g. `gitlab-test.mesosphere.com`)
+
++ Either set up appropriate certificates on each of your hosts, or configure each of your hosts to use your hostname (i.e. `gitlab-test.mesosphere.com`) as a [Docker insecure registry](https://docs.docker.com/registry/insecure/). You will need to do this for the Jenkins agent too using the [Advanced Configuration instructions](https://docs.mesosphere.com/1.8/usage/service-guides/jenkins/advanced-configuration/).
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: This is pretty confusing. Should it be a separate topic in `administration`?

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+1. Finally, you will want to ensure your DC/OS agents are authenticated against this registry. You can either run `docker login` on each of these nodes or [see the instructions](https://mesosphere.github.io/marathon/docs/native-docker-private-registry.html) on how you might distribute your Docker credentials.
+
+# Setting up the Repository
+
+1. Let's start by creating a new project in GitLab. Our application is going to use a simple customised Nginx container serving an HTML page. First, create a new project called `gitlab-demo`. For simplicity, we chose "Public" visibility.
+
+<img src="/assets/images/blog/2016-09-16-gitlab/new-project.png" alt="New GitLab Project" /> *Create a New GitLab Project*
+
+1. Once this is done, clone the project to your local machine:
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+    git clone http://gitlab-test.mesosphere.com/root/gitlab-demo.git && cd gitlab-demo
+
+1. Then, let's create a very simple index.html page:
+
+    echo "hello-world" >> index.html
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+
+1. Now, let's create a simple Dockerfile that describes how to build our Docker image:
+
+    nano Dockerfile
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+1. Paste in the following contents, which will extend the official Nginx container and add our index.html file to it:
+
+    FROM nginx
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+    COPY index.html /usr/share/nginx/html/index.html
+
+1. Next, we'll create a simple Marathon application definition. Putting this into version control is good practice and makes it easy to implement and track changes to your application definition as the underlying application:
+
+    nano marathon.json
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: nano? really? also, syntax highlighting.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+
+1. This will tell Marathon how to run our application. Paste the following into it, replacing the address to the Docker registry with your own:
+
+    {
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+        "DCOS_SERVICE_NAME": "gitlab"
+      },
+      "uris": [
+        "file:///etc/docker.tar.gz"
+      ]
+    }
+
+1. Finally, let's create a simple Jenkinsfile. This is a new way to programmatically define your builds that was released as part of [the Pipeline feature](https://jenkins.io/doc/pipeline/) of Jenkins 2.0. This will pull from the GitLab instance you've just set up, build the Docker image, push it to the GitLab registry, and then trigger a deployment using [the Marathon plugin](https://github.com/jenkinsci/marathon-plugin) for Jenkins:
+
+    nano Jenkinsfile
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+            forceUpdate: false,
+            filename: 'marathon.json',
+            appid: 'nginx',
+            docker: "gitlab-test.mesosphere.com:50000/root/gitlab-demo:${gitCommit()}".toString()
+        )
+    }
+
+1. Now that we've got our four files, add, commit and push them to the repository:
+
+    git add *
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: Syntax highlighting please.

File: src/blog/2016-09-16_gitlab-dcos.md [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+You can test this pipeline further by pushing commits to the GitLab repository. These commits will trigger new Jenkins build, which will push an updated version of the application to GitLab and then trigger a Marathon deployment!
+
+# Resources
+
+If you'd like to provide feedback or file issues against the GitLab package for DC/OS, please use [the DC/OS JIRA](https://dcosjira.atlassian.net) or send a message to the [DC/OS users list](https://groups.google.com/a/dcos.io/forum/#!forum/users)!
+
+Other resources:
+* [About GitLab](https://about.gitlab.com/about/)
+
+* [Deploy GitLab on DC/OS today](https://mesosphere.com/blog/2016/09/16/gitlab-dcos/) on the Mesosphere blog
```
Comment (pyronicide) [inline src/blog/2016-09-16_gitlab-dcos.md]: What is this link for? Feels like we're crossing the streams.

Comment (ssk2) [inline src/blog/2016-09-16_gitlab-dcos.md]: PageRank mainly. Fine.

File: src/blog/2016-09-16_gitlab-dcos.md (hunk 1 above)
Comment (ssk2) [inline src/blog/2016-09-16_gitlab-dcos.md]: My affiliation is mentioned in the author tag, so "we" was intended to refer to Mesosphere. I will make this clearer.

File: src/blog/2016-09-16_gitlab-dcos.md (hunk 2 above)
Comment (ssk2) [inline src/blog/2016-09-16_gitlab-dcos.md]: Done!

File: src/blog/2016-09-16_gitlab-dcos.md (hunk 3 above)
Comment (ssk2) [inline src/blog/2016-09-16_gitlab-dcos.md]: Done!

**Comment by ashenden** [comment]
Is this still up for review or are we good to go?

**Comment by judithpatudith** [comment]
Hey Guys, is anything happening with this post? If it goes up I'm happy to promote it on twitter etc.

**Comment by ssk2** [comment]
Blocked on me @judithpatudith.

**Comment by ssk2** [comment]
Superceded by https://github.com/dcos/examples/pull/27/

**Pull request closed without merging**